### PR TITLE
Implement rules 231XX and 232XX

### PR DESCRIPTION
**Pull request by bhavyejain**
Implemented constraints for:
CSGPrimitive
CSGSetOperator
ParametricGeometry

Modifies test files uploaded.

### CODE REVIEW

File: extensions/spatial/src/org/sbml/jsbml/validator/offline/constraints/CSGSetOperatorConstraints.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+      // identifier of an existing CSGNode object defined in the enclosing Model object.
+
+      func = new ValidationFunction<CSGSetOperator>() {
+        
+        @Override
+        public boolean check(ValidationContext ctx, CSGSetOperator csgso) {
+          if(csgso.isSetComplementA()) {
+            if(csgso.isSetListOfCSGNodes()) {
+              ListOf<CSGNode> losn = csgso.getListOfCSGNodes();
+              for(int i = 0; i < losn.getNumChildren(); i++) {
```
Comment (niko-rodrigue) [inline extensions/spatial/src/org/sbml/jsbml/validator/offline/constraints/CSGSetOperatorConstraints.java]: You should use the new form of the `for loop` here and on SPATIAL_23207:

for (CSGNode node : losn) {...}

Comment (bhavyejain) [reply extensions/spatial/src/org/sbml/jsbml/validator/offline/constraints/CSGSetOperatorConstraints.java]: Done!

**Comment by bhavyejain** [comment]
Made the suggested changes. :)

**Comment by niko-rodrigue** [comment]
Perfect, the new for loop make the code much nicer and shorter.

**Pull request merged**